spyder/widgets/variableexplorer/dataframeeditor.py

```diff
@@ -722,15 +724,28 @@ def setup_and_check(self, data, title=''):
         btn_layout.addWidget(self.bgcolor_global)
 
         btn_layout.addStretch()
-        bbox = QDialogButtonBox(QDialogButtonBox.Ok | QDialogButtonBox.Cancel)
-        bbox.accepted.connect(self.accept)
-        bbox.rejected.connect(self.reject)
-        btn_layout.addWidget(bbox)
+
+        self.btn_save_and_close = QPushButton(_('Save and Close'))
+        self.btn_save_and_close.setDisabled(True)
+        self.btn_save_and_close.clicked.connect(self.accept)
+        btn_layout.addWidget(self.btn_save_and_close)
+
+        self.btn_ok = QPushButton(_('Close'))
+        self.btn_ok.setAutoDefault(True)
+        self.btn_ok.setDefault(True)
+        self.btn_ok.clicked.connect(self.reject)
+        btn_layout.addWidget(self.btn_ok)
 
         self.layout.addLayout(btn_layout, 2, 0)
 
         return True
 
+    def save_and_close_enable(self):

```

Comment:
This function needs to receive two parameters: `top_left` and `bottom_right` that account for the indexes passed by `dataChanged`.

Reply:
Also, above this method you need to add

`@Slot(QModelIndex, QModelIndex)`